react/week3/fetchtodos/src/FetchTodos.js

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+// this componet is for styling the border
+const TodoComponent = (props) => {
+  return (
+    <div
+      style={{
+        backgroundColor: "yellow",
+        border: "2px solid black",
+        width: "700px",
+        marginLeft: "250px",
+        marginBottom: "20px",
+      }}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+const API_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function TodoList({ id, description, deadline, removeTodo }) {
+  const [isDone, setIsDone] = useState(false);
+  const [edit, setEdit] = useState(false);
+  const [newDescription, setNewDescription] = useState(description);
+
+  const updateTodo = () => {
+    setEdit(!edit);
+  };
+
+  return (
+    <>
+      <TodoComponent>
+        <div
+          className="list-of-todos"
+          style={
+            isDone ? { textDecoration: "line-through" } : { textDecoration: "" }
+          }
+        >
+          {edit ? (
+            <input
+              type="text"
+              value={newDescription}
+              onChange={(event) => {
+                const value = event.target.value;
+                setNewDescription(value);
+              }}
+            />
+          ) : (
+            newDescription
+          )}{" "}
+          | {deadline}
+          <div className="buttons">
+            <input type="checkbox" onClick={() => setIsDone(!isDone)} />
+            <button className="delet-button" onClick={() => removeTodo(id)}>
+              Delete
+            </button>
+            <button className="edit-button" onClick={updateTodo}>
+              {edit ? "Update" : "Edit"}
+            </button>
+          </div>
+        </div>
+      </TodoComponent>
+    </>
+  );
+}
+//This component is for input elements
+function RenderTodos() {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [apidata, setApiData] = useState([]);
+
+  const minimumDate = () => {
+    let today = new Date();
+    let dd = today.getDate();
+    let mm = today.getMonth() + 1; //January is 0!
+    let yyyy = today.getFullYear();
+    if (dd < 10) {
+      dd = "0" + dd;
+    }
+    if (mm < 10) {
+      mm = "0" + mm;
+    }
+
+    today = yyyy + "-" + mm + "-" + dd;
+    return today;
+  };
```

Comment:
This function does not need to be local to the `RenderTodos`-component. 
If you want to re-use it in the future, you can place it outside the component